setup.py

```diff
@@ -6,7 +6,7 @@
     long_description = fh.read()
 
 dependencies = [
-    "chia-blockchain==1.2.11",
+    "chia-blockchain>=1.2.11",
```

Comment:
this seems a bit risky to me. A lot of the chia functions used in here I would consider internal. So they may change in future versions. I would prefer if we pin a specific version and make sure the code is updated to support that.

I recognize this is an issue around upgrades, it's nice to support a beta. How about we pin this to 1.2.12 (I believe that's the unreleased version number in chia-blockchain now) without bothering about backwards compatibility?

Once the next version of chia-blockchain makes a stable release (which should be 1.3) we could bump this pinning to that and cut a new release of chia-dev-tools.

@Quexington what do you think?

Reply:
Doing a `pip install chia-dev-tools` with the dependency like this will pull the latest released version (in this case 1.2.11), which @Quexington commited to always support, and this should be fine.

If you have the trunk checked out in a venv, doing pip install chia-dev-tools will try to bring EXACTLY the pinned version into the venv with the trunk, and this is really a hot mess. You're forced to use `--no-deps`.

If you don't want `>=`, fine, then we'll have to use `--no-deps` in a venv, but at least we'll be able to use@Quexington's version with chia-blockchain trunk.

When 1.3 is released then we can remove this, change the dependency to `>= 1.3` and just read `.cost` directly.

Reply:
I'm not an expert on how dependencies are normally done in python. But generally I think our setup with the chia projects would benefit from a mono-repo setup (where all revisions move in lock-step and are always tested together).

If this project is committed to supporting multiple versions of `chia-blockchain`, then its tests need to be run against all those versions too. I defer to @Quexington

Reply:
I agree about the monorepo, folding cdv into chia-blockchain would make the most sense to me as well, but I got the impression that it is is intentionally kept in a separate repo and Quex had a reasoning for it (he wants to keep the main trunk minimal).

So I tried working respecting this decision. IIRC from Keybase, Quex said that he wants to keep it in this repo, and says that the at the moment while this package is still versionless (will be fixed in the future), it’s trunk must support the latest official release.

I try to preserve that in this commit, while still making sure it’s easy to use with the upstream trunk. >= doesn’t signify that it MUST support every future chia-blockchain release — it’s just to stop pip harassing you from installing the pinned version inside a venv with an upstream trunk.

I’m fine with keeping the pin (==) if it is your wish. I can live with having to use --no-deps.

Reply:
Yeah I think we should leave it as `==`.  `>=` will start installing the new version for everyone as soon a new release happens, which is not what we want since we need to merge PRs like this first before it is compatible with that new version.

I think this should stay its own repo, we want to move to `chia-blockchain` being mostly about consensus and full node stuff and this repo would be a move contrary to that direction.  I do think that this tool being broken with versions of chia will be a relatively rare occurrence, it just so happens that that is the state right now.  I think the best version of a workaround right now is to have 2 venvs, one with `cdv` and its pinned version of chia and one with the version of chia you want to use.

Reply:
Okay, the version pin is back per your wish. :)

Reply:
Soo, ready to merge?